### PR TITLE
Switch from `browserify` to `webpack` in order to be able to use most recent ES modules & allow JavaScript syntax beyond `ES5`

### PR DESCRIPTION
**Pull request by archmoj**
@plotly/plotly_js 

TODOs:
- [x] rewrite [`regl_codegen/devtools.js`](https://github.com/plotly/plotly.js/blob/35fb5b918e3dbdc3cc6e3b9735c8003008376020/devtools/regl_codegen/devtools.js) similar to `test_dashboard/devtools.js`.
- [x] fix `glslify` transform for regl based traces
- [x] rebuild `stackgl_modules` using webpack
- [x] do we still need to keep [`strict_d3.js`](https://github.com/plotly/plotly.js/blob/35fb5b918e3dbdc3cc6e3b9735c8003008376020/tasks/util/strict_d3.js)? Possibly no!
- [x] remove [`watchified_bundle.js`](https://github.com/plotly/plotly.js/blob/35fb5b918e3dbdc3cc6e3b9735c8003008376020/tasks/util/watchified_bundle.js).
- [x] figure out function constructor issue then revert 834d4116b35096e6215ec33164f721bb6c7a8bab.
- [x] figure out few `scattergl` test failures in `webgl-jasmine` container
- [x] update readme files
- [x] try custom bundle scripts
- [x] do not generate separate `.LICENSE.txt` files
- [x] run `compress_attributes` transform when when creating bundles other than `plotly-with-meta.js`
- [x] do not minify `plotly-with-meta.js`

### CODE REVIEW

**Comment by alexcjohnson** [comment]
@archmoj this is looking great! My biggest comment so far: can we use `webpack.config.js` as an input to [`test_dashboard/server.js`](https://github.com/plotly/plotly.js/pull/6355/files#diff-852ae8f652fcdc3aef63e08a42016fe271180244e72ab26f634db1b3a576705eR48) and [`karma.conf.js`](https://github.com/plotly/plotly.js/pull/6355/files#diff-e5d9ca9c8a09ae063f7ecd31057f8eca5882b0af32d09fa21cd280aa5436332cR263) - and just modify what we need to for those two contexts? This would be easier to maintain, clearer about what's different between the three, and give more confidence that what works in one context works in all.

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    },
+    resolve: {
+        fallback: {
+            'stream': require.resolve('stream-browserify'),
+            'buffer': require.resolve('buffer/'),
+            'assert': require.resolve('assert/')
+        }
+    },
+    plugins: [
+        new NodePolyfillPlugin({ includeAliases: ['process'] })
```
Comment (alexcjohnson) [inline webpack.config.js:46]: Did you figure out what this is needed for? Is it something in our code or in a dependency?

File: tasks/util/browserify_wrapper.js [hunk 2]
```diff
@@ -1,14 +1,10 @@
...
-var minify = require('minify-stream');
-var derequire = require('derequire');
-var through = require('through2');
+var webpack = require('webpack');
 
-var strictD3 = require('./strict_d3');
+var config = require('../../webpack.config.js');
 
-/** Convenience browserify wrapper
+/** Convenience bundle wrapper
```
Comment (alexcjohnson) [inline tasks/util/browserify_wrapper.js:7]: might want to change the filename too 😄

Comment (archmoj) [reply tasks/util/browserify_wrapper.js:7]: I would definitely change that in a separate PR to make it easier to track the changes here.

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                    ]
+                }
+            },
+        }]
+    },
+    resolve: {
+        fallback: {
+            'stream': require.resolve('stream-browserify'),
+            'buffer': require.resolve('buffer/'),
+            'assert': require.resolve('assert/')
```
Comment (alexcjohnson) [inline webpack.config.js:42]: These too - where are these needed? (It's OK if they are needed, I'm mostly just curious, also relevant for anyone who might include plotly.js from source, not dist bundles)

**Comment by archmoj** [comment]
> TODO:
> * [ ]  fix `glslify` transform for regl based traces

See https://github.com/chasevida/glslify-webpack.

File: test/jasmine/karma.conf.js [hunk 4]
```diff
@@ -260,11 +260,52 @@ func.defaultConfig = {
         }
     },
 
-    browserify: {
-        transform: [pathToStrictD3, pathToShortcutPath],
-        extensions: ['.js'],
-        watch: !argv.nowatch,
-        debug: true
+    webpack: {
+        target: ['web', 'es5'],
```
Comment (raulmarindev) [inline test/jasmine/karma.conf.js:264]: Have you considered targeting es6? https://betterprogramming.pub/do-we-still-need-to-compile-es6-javascript-code-into-es5-in-2022-f5ec70509e9b

**Comment by archmoj** [comment]
> @archmoj this is looking great! My biggest comment so far: can we use `webpack.config.js` as an input to [`test_dashboard/server.js`](https://github.com/plotly/plotly.js/pull/6355/files#diff-852ae8f652fcdc3aef63e08a42016fe271180244e72ab26f634db1b3a576705eR48) and [`karma.conf.js`](https://github.com/plotly/plotly.js/pull/6355/files#diff-e5d9ca9c8a09ae063f7ecd31057f8eca5882b0af32d09fa21cd280aa5436332cR263) - and just modify what we need to for those two contexts? This would be easier to maintain, clearer about what's different between the three, and give more confidence that what works in one context works in all.

The main part of config is reused now.

**Comment by archmoj** [comment]
Bundle sizes are slightly reduced and are looking good to my eyes.

webpack-bundle-sizes-diff

File: test/jasmine/karma.conf.js (hunk 4 above)
Comment (archmoj) [reply test/jasmine/karma.conf.js:264]: We still want to continuing support es5 syntax in our output.
But now the input syntax i.e. plotly.js source files as well as `node_modules` could be in es6 and further.

Comment (alexcjohnson) [reply test/jasmine/karma.conf.js:264]: @archmoj FWIW we just dropped IE support in Dash - and we did it in a minor https://github.com/plotly/dash/releases/tag/v2.7.0

I'd be comfortable doing the same in plotly.js, but open to discussing.

File: webpack.config.js (hunk 3 above)
Comment (archmoj) [reply webpack.config.js]: Good call. `assert` and `buffer` are no longer needed. Addressed in https://github.com/plotly/plotly.js/pull/6355/commits/9fc59870eec4138d699709d839f5f0ee39d0cdeb.

File: test/jasmine/karma.conf.js (hunk 4 above)
Comment (archmoj) [reply test/jasmine/karma.conf.js:264]: I opened https://github.com/plotly/plotly.js/issues/6366. 
Let's investigate it & decide on that separately.

File: devtools/test_dashboard/server.js [hunk 5]
```diff
@@ -3,54 +3,101 @@ var path = require('path');
...
 var open = require('open');
-var browserify = require('browserify');
+var webpack = require('webpack');
 var minimist = require('minimist');
 
 var constants = require('../../tasks/util/constants');
-var makeWatchifiedBundle = require('../../tasks/util/watchified_bundle');
-var shortcutPaths = require('../../tasks/util/shortcut_paths');
+var config = require('../../webpack.config.js');
+config.optimization = { minimize: false };
```
Comment (alexcjohnson) [inline devtools/test_dashboard/server.js:11]: Ideally even the test build would be minified, to give us further confidence we're testing the behavior of the real bundle, plus a sourcemap. Right now with no sourcemap all I see in the test dashboard is the single huge unminified bundle - all the code is there unmodified but it'd be a pain to connect it back to the source files during debugging.

Comment (archmoj) [reply devtools/test_dashboard/server.js:11]: The debugging is addressed in [458921f](https://github.com/plotly/plotly.js/pull/6355/commits/458921f44f95590e4775523f03d07f8c67d10661).
Now using *development* `mode`, the bundle content is different from *production* mode; but it builds & rebuilds faster.
This should be good for now. And we could further improve this configuration.

Comment (alexcjohnson) [reply devtools/test_dashboard/server.js:11]: That'll work - what you see in the debugger isn't quite the same as the source files - it's been reformatted a bit and the require statements are transformed - but it's pretty close and easy to move from one to another 😄

**Review by alexcjohnson — APPROVED** [review]
💃 Let's do it!

**Pull request merged**